client/components/overview/OverviewContainer.jsx

```diff
@@ -20,9 +20,25 @@ const OverviewContainer = () => {
   const [productInfo, setProductInfo] = useState({});
 
   useEffect(() => {
-    getProductInfo(currentProduct, setProductInfo);
     getProductStyles(currentProduct, setProductStyles);
-  }, []);
+    getProductInfo(currentProduct, setProductInfo);
+  }, [currentProduct]);
+
+  useEffect(() => {
+    if (productStyles.length > 1) {
+      const defaultStyle = () => {
+        const initialStyle = productStyles.filter((style) => {
+          if (style.default === true) {
```

Comment:
A shorter way to write this would be `productStyles.filter((style) => style.default)` since `style.default` evaluates to `true` or `false`. I'm not sure if that would be good practice though & maybe it's a good idea to explicitly write `productStyles.filter((style) => style.default === true)` instead. No fix needed, just something to consider.